packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -339,6 +339,23 @@ export class PHPRequestHandler {
 
 		let fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
 
+		// If we are dealing with a symlink, lets resolve the target
+		// and continue using that instead of the symlink.
+		if (primaryPhp.isSymlink(fsPath)) {
+			const originalPathEndsWithSlash = fsPath.endsWith('/');
+			fsPath = primaryPhp.readlink(fsPath);
```

Comment:
Maybe we should implement a `realpath()` method and use that so that symlinks to symlinks are resolved completely.

In addition, let's only overwrite `fsPath` if the symlink resolves successfully and let symlink resolution failures fall through to [the file-not-found handling callback](https://github.com/WordPress/wordpress-playground/blob/2975b41ced95054ab3332cad41f27af557309b08/packages/php-wasm/universal/src/lib/php-request-handler.ts#L383-L386).

Reply:
I've implemented a rough version of a realpath function. It looks a bit complicated but this is mainly because I also wanted to support paths that have a symlink in the middle. We need to iterate over the parts of the path and check if its a symlink and if we find a symlink, we repeat the process for the target of the symlink. 

I investigated if we need to also consider symlinks that contain relative paths like `/../../` but it seems that once you create a symlink on NODEFS, these paths are already resolved to absolute paths. The same goes for when you mount a symlink or directory with a symlink in it. In other words, i was not able to create a symlink with a relative path component so it seems we don't need to support it.

Any other cases we need to consider?

Reply:
Thanks for adding the realpath() implementation! I will try to review that tomorrow but am out of time today.

Reply:
Seeing the realpath() implementation replaced by a lookupPath() call is great!